funding/manager_test.go

```diff
@@ -4733,3 +4755,206 @@ func TestFundingManagerZeroConf(t *testing.T) {
+		Updates:         updateChan,
+		Err:             errChan,
+	}
+
+	alice.fundingMgr.InitFundingWorkflow(initReq)
+
+	// Alice should send an OpenChannel to Bob.
+	var aliceMsg lnwire.Message
+	select {
+	case aliceMsg = <-alice.msgChan:
+	case err := <-initReq.Err:
+		t.Fatalf("error init funding workflow: %v", err)
+	case <-time.After(time.Second * 5):
+		t.Fatalf("alice did not send OpenChannel message")
+	}
+
+	openChannelReq, ok := aliceMsg.(*lnwire.OpenChannel)
+	require.True(t, ok)
+
+	bob.fundingMgr.ProcessFundingMsg(openChannelReq, alice)
+
+	// Bob should answer with an AcceptChannel message.
+	acceptChannelResponse, ok := assertFundingMsgSent(
+		t, bob.msgChan, "AcceptChannel",
+	).(*lnwire.AcceptChannel)
+	require.True(t, ok)
+
+	// Forward the response to Alice.
+	alice.fundingMgr.ProcessFundingMsg(acceptChannelResponse, bob)
+
+	// Alice responds with a FundingCreated message.
+	fundingCreated, ok := assertFundingMsgSent(
+		t, alice.msgChan, "FundingCreated",
+	).(*lnwire.FundingCreated)
+	require.True(t, ok)
+
+	// Give the message to Bob.
+	bob.fundingMgr.ProcessFundingMsg(fundingCreated, alice)
+
+	// Finally, Bob should send the FundingSigned message.
+	fundingSigned, ok := assertFundingMsgSent(
+		t, bob.msgChan, "FundingSigned",
+	).(*lnwire.FundingSigned)
+	require.True(t, ok)
+
+	// Forward the signature to Alice.
+	alice.fundingMgr.ProcessFundingMsg(fundingSigned, bob)
+
+	var pendingUpdate *lnrpc.OpenStatusUpdate
+	select {
+	case pendingUpdate = <-updateChan:
+	case <-time.After(time.Second * 5):
+		t.Fatalf("alice did not send OpenStatusUpdate_ChanPending")
+	}
+
+	_, ok = pendingUpdate.Update.(*lnrpc.OpenStatusUpdate_ChanPending)
+	require.True(t, ok)
+
+	// Confirm the funding transaction.
+	alice.mockNotifier.oneConfChannel <- &chainntnfs.TxConfirmation{
+		Tx: fundingTx,
+	}
+
+	bob.mockNotifier.oneConfChannel <- &chainntnfs.TxConfirmation{
+		Tx: fundingTx,
+	}
+
+	// Make sure the notification about the pending channel was sent out.
+	select {
+	case <-alice.mockChanEvent.pendingOpenEvent:
+	case <-time.After(time.Second * 5):
+		t.Fatalf("alice did not send pending channel event")
+	}
+	select {
+	case <-bob.mockChanEvent.pendingOpenEvent:
+	case <-time.After(time.Second * 5):
+		t.Fatalf("bob did not send pending channel event")
+	}
+
+	// Assert that neither alice nor bob sees the channel as open yet. This
+	// is because the coinbase check is hit and we must wait for more
+	// confirmations.
+	select {
+	case <-alice.mockChanEvent.openEvent:
+		t.Fatalf("alice sent an open channel event")
+	case <-time.After(time.Second * 5):
+	}
+
+	select {
+	case <-bob.mockChanEvent.openEvent:
+		t.Fatalf("bob sent an open channel event")
+	case <-time.After(time.Second * 5):
+	}
+
+	// Send along the oneConfChannel again and then assert that the open
+	// event is sent. This serves as the 100 block + MinAcceptDepth
+	// confirmation.
+	alice.mockNotifier.oneConfChannel <- &chainntnfs.TxConfirmation{
+		Tx: fundingTx,
+	}
```

Comment:
This mock test does exercise the new code, but it's brittle.  It does not actually test that coinbase maturity has been reached or that the channel can be closed immediately after `channel_ready`.

I think an itest is needed to properly test what we want.

Reply:
To test this in an itest, this needs to use the psbt flow and bitcoind as a miner to generate directly to the p2wsh via `generatetoaddress`. However, we use btcd as itest miner which doesn't have a `generatetoaddress` analog. We can't stop btcd and change the `--miningaddr` flag because the itests are parallel and another test may be using the miner. This unit test isn't ideal, but does give some coverage. I have also tested this on my machine several times with psbt's and bitcoind and verified that the channel was unusable until after 100+min depth confs. Unfortunately, I don't have time to change btcd's code to add the `generatetoaddress` rpc so I think this is the best we can do right now